cmd/podman/main.go

```diff
@@ -40,8 +40,16 @@ func main() {
 
 	if filepath.Base(os.Args[0]) == registry.PodmanSh ||
 		(len(os.Args[0]) > 0 && filepath.Base(os.Args[0][1:]) == registry.PodmanSh) {
+		// podmansh is only meant to be set as a login shell for non-root
+		// users. So, there is no need to enable it for root.
+		if !rootless.IsRootless() {
+			os.Exit(125)
+		}
 		shell := strings.TrimPrefix(os.Args[0], "-")
-		args := []string{shell, "exec", "-i", "--wait", "10"}
+
+		// The wait timeout will soon be made configurable via the
+		// upcoming `podmansh_timeout` option in containers.conf
+		args := []string{shell, "exec", "-i", "--wait", "30"}
```

Comment:
This should be read from containers.conf?

Reply:
getting from containers.conf would need c/common vendoring. That means this can't go into v4.6.0.

Reply:
if we're good to cut a new version of c/common and assuming we're ok with this not making v4.6.0, I can vendor it in this PR

Reply:
lets get this in, then as follow up you can vendor c/common to latest and use the field on main